src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -426,9 +430,10 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                             }
                         }
 
-                        if (responsePaths != null)
+                        if (responsePaths != null &&
+                            PathUtilities.GetDirectoryName(resolvedPath) is string directory)
```

Comment:
>PathUtilities.GetDirectoryName(resolvedPath) is string directory [](start = 28, length = 64)

Why add this condition and not just assert the result is not null in the body of the `if`? This potentially changes evaluation semantics.

Reply:
If we did that, I think we'd hit the assertion. I'm not 100% though (didn't manage to hit it in test, but this may be platform-dependent...)

---
In reply to: [377822593](https://github.com/dotnet/roslyn/pull/41063#discussion_r377822593) [](ancestors = 377822593)